File: app/src/main/java/com/example/footmatch/util/api/ApiKeyInterceptor.java
package com.example.footmatch.util.api;

import androidx.annotation.NonNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;
public class ApiKeyInterceptor implements Interceptor {
    private static final String API_KEY = "e1718a84ac1a4301a685e83a5a8d227a";

    @NonNull
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request originalRequest = chain.request();
        Request newRequest = originalRequest.newBuilder()
                .addHeader("X-Auth-Token", API_KEY)
                .build();

        return chain.proceed(newRequest);
    }
}

File: app/src/main/java/com/example/footmatch/datos/api/ApiKeyInterceptor.kt
package com.example.footmatch.datos.api

import okhttp3.Interceptor

import okhttp3.Request
import okhttp3.Response
import java.io.IOException

class ApiKeyInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val originalRequest: Request = chain.request()
        val newRequest = originalRequest.newBuilder()
            .addHeader("X-Auth-Token", API_KEY)
            .build()
        return chain.proceed(newRequest)
    }

    companion object {
        private const val API_KEY = "e1718a84ac1a4301a685e83a5a8d227a"
    }
}